frame/support/src/storage/bounded_vec.rs

```diff
@@ -222,16 +222,17 @@ impl<T, S: Get<u32>> BoundedVec<T, S> {
 	/// Returns `true` if the item was inserted.
 	pub fn force_insert_keep_left(&mut self, index: usize, element: T) -> bool {
 		// Check against panics.
-		if Self::bound() < index || self.len() < index {
+		if Self::bound() < index || self.len() < index || Self::bound() == 0 {
```

Comment:
This makes sense for readability, but I think this condition is already covered assuming the vec is in a valid state. 

Say `Self::bound() ==  0`, then either the user passes index 0 and length is 0, meeting the below condition `Self::bound() == index && self.len() <= Self::bound()`
Or
`index > 0` and we meet the condition `Self::bound() < index`

Reply:
yup, i had saw this too, but i just think this is better for being explicit.